### DISCUSSION TITLE
webui very slow with P40 - 1,6 token/s with TheBloke_vicuna-13B-1.1-GPTQ-4bit-128g

### DISCUSSION THREAD

**Discussion by gandolfi974** [oobabooga/text-generation-webui#2311] [Q&A]
hello,
i'm beginner with text-generation webui. I find it very slow with 1,6 token/s for the model "THE BLOKE - vicuna-13B-1.1-GPTQ-4bit-128g.latest.safetensors".
i have a ryzen 5 2400g + Tesla P40 + 48 go Ram on ubuntu 22.04.

with nvidia-smi command :
`NVIDIA-SMI 530.30.02              Driver Version: 530.30.02    CUDA Version: 12.1    `

i have tested with or without xformers but no difference.
`python server.py --auto-devices --xformers`

- do you think is normal ?
- are there any particular settings or drivers ?

thanks

**Comment by cibernicola** [comment]
I've tested several models from TheBloke, average t/s is around 10. RTX 3090/ Ryzen 5 3600XT / 48GB RAM / Win10 / xformers

**Reply by gandolfi974** [reply]
- Is it 13b or 30b ? Gptq ?
- what software and configuration are you using.
 I tried to understand if this slowness comes from the p40, Ubuntu or the settings.
Thanks

**Reply by cibernicola** [reply]
I'm using webUI with TheBloke_Wizard-Vicuna-13B-Uncensored-GPTQ,, python server.py --auto-devices --chat --threads 150  --api --bf16 --xformers  (not sure if threads flag still working), under Windows10
![image](https://github.com/oobabooga/text-generation-webui/assets/4579387/cfe95ce2-93be-4345-80e3-0dfa4baba21c)

This is the result: Output generated in 17.97 seconds (11.07 tokens/s, 199 tokens, context 53, seed 832234523)

**Reply by gandolfi974** [reply]
thanks. after a lot of try, i have Output generated in 10.20 seconds (9.22 tokens/s, 94 tokens, context 48, seed 292081768)
Fast_kernel must be "False" with "True" i have only 1 token/s

**Reply by silenceyese** [reply]
hi, I have the same problem like yours. Can u tell me how to set Fast_kernel to "False"? thanks very much~
